project/webapp/controller/ProductsList.controller.js

```diff
@@ -1,15 +1,193 @@
-sap.ui.define([
-    "sap/ui/core/mvc/Controller"
-],
-    /**
-     * @param {typeof sap.ui.core.mvc.Controller} Controller
-     */
-    function (Controller) {
-        "use strict";
+sap.ui.define(
+  [
+    "sap/ui/core/mvc/Controller",
+    "sap/ui/model/json/JSONModel",
+    "veronchi/leverx/project/model/productModel",
+    "veronchi/leverx/project/model/filterBarModel",
+    "sap/ui/model/Filter",
+    "sap/ui/model/FilterOperator",
+  ],
 
-        return Controller.extend("veronchi.leverx.project.controller.ProductsList", {
-            onInit: function () {
+  function (Controller, JSONModel, productModel, filterBarModel, Filter, FilterOperator) {
+    "use strict";
 
-            }
+    return Controller.extend("veronchi.leverx.project.controller.ProductsList", {
+      APP_MODEL_NAME: "appModel",
+      TABLE_MODEL_NAME: "tableModel",
+      FILTER_BAR_MODEL_NAME: "filterBarModel",
+      TOKEN_REMOVED_TYPE: "removed",
+
+      onInit() {
+        productModel.initModel();
+        filterBarModel.initFilterBarModel();
+        const oModel = productModel.getModel();
+        this.oFilterBarModel = filterBarModel.getFilterBarModel();
+
+        this.oTableModel = new JSONModel({
+          isProductsSelected: false,
+        });
+
+        this.getView().setModel(oModel, this.APP_MODEL_NAME);
+        this.getView().setModel(this.oTableModel, this.TABLE_MODEL_NAME);
+        this.getView().setModel(this.oFilterBarModel, this.FILTER_BAR_MODEL_NAME);
+      },
+
+      onSelectProduct(bProductSelected) {
+        this.oTableModel.setProperty("/isProductsSelected", bProductSelected);
+      },
+
+      onSearchProducts(oEvent) {
+        const oTableBinding = this.byId("productList").getBinding("items");
+        const aFilters = this._getAllFilters(oEvent);
+
+        oTableBinding.filter(aFilters);
+      },
+
+      onClearFilters() {
+        const oTableBinding = this.byId("productList").getBinding("items");
+
+        this.byId("releaseDate").setValue(null);
+        this.byId("supplier").setTokens([]);
+        this.byId("searchName").setValue(null);
+        this.byId("categorySelect").setSelectedKeys([]);
+
+        oTableBinding.filter(null);
+      },
+
+      _getSearchNameFilter() {
+        const sSearchQuery = this.byId("searchName").getProperty("value");
+
+        return sSearchQuery.length ? new Filter("name", FilterOperator.Contains, sSearchQuery) : null;
+      },
+
+      _getCategoriesFilter() {
+        const aSelectedCategories = this.byId("categorySelect").getProperty("selectedKeys");
+        const aFilters = [];
+
+        if (!!aSelectedCategories.length) {
+          aSelectedCategories.map((sSelectedKey) => {
```

Comment:
it's more convenient to do that like
aFilters =  aSelectedCategories.map(sSelectedKey => new Filter(....))
to get rid off aFilters.push